spyder/plugins/editor.py

```diff
@@ -1887,14 +1887,47 @@ def save(self, index=None, force=False):
         return editorstack.save(index=index, force=force)
     
     @Slot()
-    def save_as(self):
-        """Save *as* the currently edited file"""
+    def save_as(self, index=None):
+        """Save file as..."""
+
         editorstack = self.get_current_editorstack()
-        if editorstack.save_as():
-            fname = editorstack.get_current_filename()
-            if CONF.get('workingdir', 'editor/save/auto_set_to_basedir'):
-                self.open_dir.emit(osp.dirname(fname))
-            self.__add_recent_file(fname)
+        #results = editorstack.get_todo_results()
+
+        if index is None:
+            index = editorstack.get_stack_index()
+
+        finfo = editorstack.data[index]

```

Comment:
Why is all this code here? There was a save_as method already on the editor widget, located at 
https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/editor.py#L1310

Why not make the changes there?